vortex-array/src/compute/take.rs

```diff
@@ -77,12 +77,6 @@ pub fn take(array: impl AsRef<Array>, indices: impl AsRef<Array>) -> VortexResul
         "Take length mismatch {}",
         array.encoding()
     );
-    debug_assert_eq!(
```

Comment:
If take codes can be nullable then the resulting array can be nullable after perfoming the take even if it wasn't. We also check the dtype always. Happy to switch it up and remove the release dtype check